R/utils-code-dependency.R

```diff
@@ -0,0 +1,439 @@
+  symbol_effects_names <-
+    unlist(
+      lapply(
+        pd,
+        function(x) {
+          com_cond <-
+            x$token == "COMMENT" & grepl("@effect", x$text) & grepl(paste0("[\\s]*", object, "[\\s$]*"), x$text)
+
+          # Make sure comment id is not the highest id in the item.
+          # For calls like 'options(prompt = ">") # @effect ADLB',
+          # 'options(prompt = ">")' is put in a one item
+          # and '# @effect ADLB' is the first element of the next item.
+          # This is tackled in B.
+
+
+          if (!com_cond[1] & sum(com_cond) > 0) {
+            # A.
+            x[x$token == "SYMBOL", "text"]
+          } else if (com_cond[1] & sum(com_cond[-1]) > 0) {
+            # B.
+            x <- x[-1, ]
+            x[x$token == "SYMBOL", "text"]
+          }
+        }
+      )
+    )
+
+  commented_calls <- vapply(pd,
+    function(x) any(x$token == "COMMENT" & grepl("@effect", x$text)),
+    FUN.VALUE = logical(1)
+  )
+
+  symbol_effects_lines <-
+    unlist(
+      lapply(
+        symbol_effects_names,
+        function(x) {
+          code <- return_code(x, pd = pd, occur = occur, cooccur = cooccur)
+          # QUESTION: SHOULD cooccur BE TRIMMED like it happens in return_code()?
+          if (is.null(code)) {
+            # Extract lines for objects that were used, but never created.
+            # Some objects like 'iris' or 'mtcars' are pre-assigned in the session.
+            # Below is just used for comments with @effect.
+            # if (!object %in% names(occur)) {
+            intersect(which(detect_symbol(x, pd)), which(commented_calls))
+            # }
+          } else {
+            code
+          }
+        }
+      )
+    )
+
+  # When commet_id is the highest id in the item - take previous item.
+  side_effects_names <-
+    unlist(
+      lapply(
+        pd,
+        function(x) {
+          com_cond <-
+            x$token == "COMMENT" & grepl("@effect", x$text) & grepl(paste0("[\\s]*", object, "[\\s$]*"), x$text)
+
+          # Work out the situation when comment id is the highest id in the item.
+          # For calls like 'options(prompt = ">") # @effect ADLB',
+          # 'options(prompt = ">")' is put in a one item
+          # and '# @effect ADLB' is the first element of the next item.
+
+          com_cond[1]
+        }
+      )
+    )
+
+  side_effects_lines <- which(side_effects_names) - 1
+
+  sort(unique(c(symbol_effects_lines, side_effects_lines)))
+}
+
+#' Return the lines of code (with side-effects) needed to reproduce the object.
+#' @return `character` vector of elements of `parsed_code` calls that were required to build the side-effects and
+#' influencing objects having and impact on the `object`
+#'
+#' @param object `character` with object name
+#' @param parsed_code (`expression`) result of `parse()`
+#' @keywords internal
+get_code_dependencies <- function(qenv, name) {
```

Comment:
This should be a private function implemented in `get_code` - so `get_code` should also have `name` argument with `NULL` as default (get me all code)

Reply:
Hey @gogonzo, very good idea. I made a change in this commit 563265b90920c2d46f61de4de723a0efbf31ce45 
I am wondering if we could change the order of parameters in `get_code` so `name` is the second argument as it will be used more frequently than `deparse`. However I put name as 3rd parameter to maintain reproducibility for the current usage, where `deparse` could be used as a second argument, without being named.

Reply:
It actually needs to be called with `deparse = FALSE` like `get_code(q2, deparse = FALSE, name = "ADLB")` 0988658e24c0d4af4c8956b8d688a0610d371b24